module/builder/consensus/builder.go

```diff
@@ -282,7 +281,7 @@ func (b *Builder) getInsertableGuarantees(parentID flow.Identifier) ([]*flow.Col
 	// look up the root height so we don't look too far back
 	// initially this is the genesis block height (aka 0).
 	var rootHeight uint64
-	err = b.db.View(operation.RetrieveRootHeight(&rootHeight))
+	err = operation.RetrieveRootHeight(b.db.Reader(), &rootHeight)
```

Comment:
TODO: consider reading the root height from the protocol state, so that we can get rid of this dependency on the database. 

Replace with `state.Params().FinalizedRoot()`.